pkg/pgxconn/implement.go

```diff
@@ -0,0 +1,105 @@
+package pgxconn
+
+import (
+	"time"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgproto3/v2"
+	"github.com/jackc/pgx/v4"
+)
+
+// rowWithMetric wraps the row returned by QueryRow() for metrics telemetry.
+type rowWithMetric struct {
+	r pgx.Row
+}
+
+func (w rowWithMetric) Scan(dest ...interface{}) error {
```

Comment:
nit: golang best practices require comment with description for every public function. More in https://go.dev/blog/godoc

Reply:
This is an interface implementation, so these functions will never be visible when hovered in an IDE.